### ISSUE TITLE
Support `--ignoreStatic`

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker-js#3282]
We want StrykerJS to be able to ignore static mutants. This is a requirement for supporting native ES modules, as well as a performance booster for others.

See #2922

**Comment by nicojs** [comment]
Work on `--ignoreStatic` has started in the [`feat/ignore-static`](https://github.com/stryker-mutator/stryker-js/tree/feat/ignore-static) branch.

The preliminary results are great! Tests with running Stryker on `@stryker-mutator/core`: 

Without ignoring static:
```
INFO MutationTestExecutor Done in 8 minutes 50 seconds.
```

With `--ignoreStatic`
```
INFO MutationTestExecutor Done in 3 minutes 56 seconds.
```
_Note: I've disabled the typescript-checker for this test run_

The difference is 189 mutants out of a total of 3115. So ~6% of all mutants cost more than 50% of the execution time. Absolutely mindblowing 🤯

The resulting mutation score is different (as to be expected).

**Comment by nicojs** [comment]
Closed with #3284
